html/feed/feed.js

```diff
@@ -1,5 +1,369 @@
+import timeStamp from "../utils/helpers/timeStamp.js";
+import postComment from "../utils/helpers/postComment.js";
+import reactToPosts from "../utils/helpers/reactToPosts.js";
+import deletePost from "../utils/helpers/deletePost.js"; 
+import editPost from "../utils/helpers/editPost.js"; 
+
+const feedContainer =document.querySelector ("#feed-container")
+const confirmEditBtn = document.querySelector ("#confirmEdit")
+
+let token 
 const baseUrl = "https://api.noroff.dev/api/v1";
 
+let userName 
+token = localStorage.getItem ('bearerToken',token); 
+userName = localStorage.getItem ('name',userName); 
+
+const options = {
+  method: "GET", 
+  credentials: "same-origin", 
+  headers: {
+    "Content-Type": "application/json", 
+    Authorization: `Bearer ${token}`
+  }
+}
+
+
+async function getPosts (headerOptions){
+  try {
+      const response = await fetch (`${baseUrl}/social/posts?_author=true&_comments=true&_reactions=true`,headerOptions); 
+      const posts= await response.json();
+      console.log (posts); 
+      return posts;
+  } catch (error){
+      console.log (error);
+  }
+}
+
+
+async function generatePage (){
+  const data = await getPosts (options)
+  if (data){
+    generateProfileCards (data, feedContainer, userName)
+  }
+}
+
+
+confirmEditBtn.addEventListener('click', async function(event){
+
+  const postId = event.currentTarget.getAttribute ('data-post-id');
+  let modal = bootstrap.Modal.getInstance (document.querySelector ('#editPostModal'));
+  let updatedTitle= document.querySelector ('#editTitle').value;
+  let updatedBody = document.querySelector ('#editBody').value; 
+
+  const response = await editPost (baseUrl, postId, updatedTitle, updatedBody, token); 
+  console.log (response)
+
+  if (response.updated > resposne.created){
+    const postCard = document.getElementById (postId); 
+    const titleElement = postCard.querySelector ('span'); 
+    const bodyElement = postCard.querySelector ('.card-text'); 
+
+    titleElement.textContent= updatedTitle; 
+    bodyElement.textContent = updatedBody;
+    updatedTitle =""; 
+    updatedBody = ""; 
+    modal.hide (); 
+  }
+});
+
+
+generatePage ()
+
+
+function generateProfileCards (data,container, userProfile){
+
+
+  data.forEach(element => {
+
+    const card = document.createElement ('div'); 
+    card.className = 'card mb-3 w-100 p-2'; 
+    card.id = element.id;
+
+    const row = document.createElement ('div'); 
+    row.className = ' row g-0'; 
+
+    const colImage = document.createElement ('div'); 
+    colImage.className = 'col-md-auto py-2 pl-3';
+
+    const userImg = document.createElement ('div'); 
+    userImg.src = element.author.avatar ?? "../../img/lion2.jpg"; 
```

Comment:
this wont work with a div. only img tags can have src.
please update so the the div wraps around a img element.

Reply:
fixed